zeppelin-web/src/app/visualization/builtins/visualization-nvd3chart.js

```diff
@@ -37,14 +37,39 @@ export default class Nvd3ChartVisualization extends Visualization {
       this.chart = nv.models[type]();
     }
 
+    // prevent to overlap in text if using multiBarChart
+    // if size of xLabel are over maxRotate, the MutiBarChart don't show text
+    if (type === 'multiBarChart' && d3g[0]) {
```

Comment:
Could you try place this routine inside of https://github.com/apache/zeppelin/blob/master/zeppelin-web/src/app/visualization/builtins/visualization-barchart.js#L58 ?